doc/guides/diagnostic-tooling-support-tiers.md

```diff
@@ -0,0 +1,141 @@
+# Diagnostic Tooling Support Tiers
+
+Diagnostic tooling is important to the consumers of Node.js. It is used both
+in development and in production in order to investigate problems.  The failure
+of one of these tools may be as big a problem for an end user as a bug within
+the runtime itself.
+
+The Node.js project has assessed the tools and the APIs which support those
+tools. Each of the tools and APIs has been put into one of
+the following tiers.  
+
+* Tier 1 - Must always be working(CI tests passing) for all
+  Current and LTS Node.js releases. A release will not be shipped if the test
+  suite for the tool/API is not green. To be considered for inclusion
+  in this tier it must have a good test suite and that test suite and a job
+  must exist in the Node.js CI so that it can be run as part of the release
+  process.  No commit to master should break this tool/API if the next
```

Comment:
Maybe we should be more specific about what `No commit to master should break this tool/API` means? We probably will not run the tests of these tools for every PR to master, so this will be hard to enforce as-is. We may either:

1. Run the tests of these tools with master every day/week and revert any recent commit that breaks the tests
2. Acknowledge that we may have commits on master that break the tools without being noticed (because we may not run the tooling tests for those PRs), and limit this to `no commit to release branches should break...`, which means we will run the tooling tests in the release process and exclude commits that break the tests (optionally, run the tests when someone suspects a PR may break the tools, i.g. similar to how CITGM works). This would be more practical.

Reply:
I'm ok with 2, given you running for every commit is problematic.  In addition to the release process we should still aim to run them nightly if possible so that we get earlier warning if possible.